src/document/document.ts

```diff
@@ -318,25 +316,26 @@ export class Document<T> {
           }
 
           const changeInfos: Array<ChangeInfo> = [];
-          for (const { message, operations, actor } of event.value) {
-            const targetOps: Array<OperationInfo> = [];
-            for (const op of operations) {
-              if (this.isSameElementOrChildOf(op.path, target)) {
-                targetOps.push(op);
-              }
+          const { message, operations, actor } = event.value;
+          const targetOps: Array<OperationInfo> = [];
+          for (const op of operations) {
+            if (this.isSameElementOrChildOf(op.path, target)) {
+              targetOps.push(op);
             }
-            targetOps.length &&
-              changeInfos.push({
-                message,
-                operations: targetOps,
-                actor,
-              });
           }
-          changeInfos.length &&
```

Comment:
The variable `changeInfos` is not necessary.

```ts
targetOps.length &&
  callback({
    type: event.type,
    value: {
      message,
      operations: targetOps,
      actor,
    },
  });
```

Reply:
good comment